source/common/upstream/edf_scheduler.h

```diff
@@ -113,18 +113,19 @@ template <class C> class EdfScheduler : public Scheduler<C> {
     //
     // The following code does P picks, by first emulating p'_i picks for each
     // entry, and then executing the leftover P - sum(p'_i) picks.
-    double weights_sum = std::accumulate(
-        entries.cbegin(), entries.cend(), 0.0,
-        [&aug_calculate_weight](double sum_so_far, const std::shared_ptr<C>& entry) {
-          return sum_so_far + aug_calculate_weight(*entry);
-        });
+    std::vector<double> weights;
+    weights.reserve(entries.size());
+    std::transform(entries.cbegin(), entries.cend(), std::back_inserter(weights),
+                   [&aug_calculate_weight](const std::shared_ptr<C>& entry) {
+                     return aug_calculate_weight(*entry);
+                   });
```

Comment:
Please move this code to be before the comment above, and please add a comment explaining why is this desired over accessing `aug_calculate_weight` directly.

Reply:
Done.